src/movies.js

```diff
@@ -1,26 +1,80 @@
 // The `movies` array from the file `src/data.js`.
-console.log('movies: ', movies);
-
+// console.log('movies: ', movies);
 
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+
+function getAllDirectors(movies) {
+  const directors = movies.map(item => {
+    return item.director;
+  });
+  return directors;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(movies) {
+  const times = movies.filter(item => item.director === 'Steven Spielberg');
+return times.length;
+}
```

Comment:
Here you are asked to filter by two factors, by director and by genre , so missing one more check ;)

Reply:
I see. Thanks!